src/System.Runtime/tests/System/Runtime/CompilerServices/RuntimeHelpersTests.cs

```diff
@@ -110,6 +113,70 @@ internal class HasCctorReceiver
         {
             public static string S;
         }
+
+        [Fact]
+        public static void PrepareMethod()
+        {
+            foreach (MethodInfo m in typeof(RuntimeHelpersTests).GetMethods())
+                RuntimeHelpers.PrepareMethod(m.MethodHandle);
+
+            Assert.Throws<ArgumentException>(() => RuntimeHelpers.PrepareMethod(default(RuntimeMethodHandle)));
+            Assert.Throws<ArgumentException>(() => RuntimeHelpers.PrepareMethod(typeof(IList).GetMethod("Add").MethodHandle));
+        }
+
+        [Fact]
+        public static void PrepareGenericMethod()
+        {
+            Assert.Throws<ArgumentException>(() => RuntimeHelpers.PrepareMethod(default(RuntimeMethodHandle), null));
+
+            //
+            // Type instantiations
+            //
+
+            // Generic definition with instantiation is valid
+            RuntimeHelpers.PrepareMethod(typeof(List<>).GetMethod("Add").MethodHandle, 
+                new RuntimeTypeHandle[] { typeof(TestStruct).TypeHandle });
+
+            // Instantiated method without instantiation is valid
+            RuntimeHelpers.PrepareMethod(typeof(List<int>).GetMethod("Add").MethodHandle,
+                null);
+
+            // Generic definition without instantiation is invalid
+            Assert.Throws<ArgumentException>(() => RuntimeHelpers.PrepareMethod(typeof(List<>).GetMethod("Add").MethodHandle,
+                null));
+
+            // Wrong instantiation
+            Assert.Throws<ArgumentException>(() => RuntimeHelpers.PrepareMethod(typeof(List<>).GetMethod("Add").MethodHandle,
+                new RuntimeTypeHandle[] { typeof(TestStruct).TypeHandle, typeof(TestStruct).TypeHandle }));
+
+            //
+            // Method instantiations
+            //
+
+            // Generic definition with instantiation is valid
+            RuntimeHelpers.PrepareMethod(typeof(Array).GetMethod("Resize").MethodHandle,
+                new RuntimeTypeHandle[] { typeof(TestStruct).TypeHandle });
+
+            // Instantiated method without instantiation is valid
+            RuntimeHelpers.PrepareMethod(typeof(Array).GetMethod("Resize")
+                    .MakeGenericMethod(new Type[] { typeof(TestStruct) }).MethodHandle,
+                null);
+
+            // Generic definition without instantiation is invalid
+            Assert.Throws<ArgumentException>(() => RuntimeHelpers.PrepareMethod(typeof(Array).GetMethod("Resize").MethodHandle,
+                null));
+
+            // Wrong instantiation
+            Assert.Throws<ArgumentException>(() => RuntimeHelpers.PrepareMethod(typeof(Array).GetMethod("Resize").MethodHandle,
+                new RuntimeTypeHandle[] { typeof(TestStruct).TypeHandle, typeof(TestStruct).TypeHandle }));
+        }
+
+        [Fact]
+        public static void PrepareDelegate()
+        {
+            RuntimeHelpers.PrepareDelegate((Action)(() => { }));
+            RuntimeHelpers.PrepareDelegate(null);
+        }
```

Comment:
I know there was a comment in the implementation about multicast delegates and how each constituent delegate needs to be prepared individually.  Even so, maybe worth validating that nothing blows up when providing multiple delegates?